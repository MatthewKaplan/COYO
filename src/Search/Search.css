.searchStudioContainer {
  display: flex;
  height: 40px;
  margin: 0 auto;
  padding: 0.5rem;
  width: 275px;
}

.searchStudio {
  display: inline-flex;
  flex: 1 1 400px;
  overflow: hidden;
  position: relative;
}

.searchIcon {
  padding: 0.2rem;
  font-size: 16px;
}

.studioSearch {
  border: 1px solid #538ac5;
  flex: 1;
  font-size: 16px;
  padding: 0.2rem 0.2rem 0.2rem 1rem;
}

.studioSearch:focus {
  border-right: none;
  border: 1px solid black;
  outline: none;
}