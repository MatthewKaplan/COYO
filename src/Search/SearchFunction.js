import data from '../data.js';

export default function search(searchQuery) {
  const results = [];
  let query = searchQuery.toLowerCase();

  let matchingStudios = data.studios.filter(studio => studio.name.toLowerCase().includes(query));
  matchingStudios.forEach(match => results.push(match));

  return results;
}