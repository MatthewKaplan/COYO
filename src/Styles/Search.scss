@import 'Mixins';

.searchStudioContainer {
  display: flex;
  flex-direction: column;
  height: 40px;
  margin: 0 auto;
  // padding: 0.5rem;
  width: 275px;
}

.searchStudio {
  display: inline-flex;
  // flex: 1 1 400px;
  // overflow: hidden;
  // position: relative;
}

.searchIcon {
  padding: 0.2rem;
  font-size: 16px;
}

.studioSearch {
  border: 1px solid $studio-search-border;
  border-radius: 4px;
  flex: 1;
  font-size: 16px;
  padding: 0.2rem 0.2rem 0.2rem 1rem;
}

.studioSearch:focus {
  border-right: none;
  border: 1px solid $studio-search-focus;
  outline: none;
}

//everything below this line needs to be moved into Controls.scss
.btnContainer {
  display: flex;
  flex-direction: column;
  margin: 10px;
}

.showAll {
  background-color: $show-all-btn;
  border-radius: 4px;
  color: $show-btn-text;
  margin-bottom: 10px;
}

.sort {
  background-color: $show-all-btn;
  border-radius: 4px;
  color: $show-btn-text;
}