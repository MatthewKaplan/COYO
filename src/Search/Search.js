import React, { Component } from "react";
import '../Styles/Search.scss';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchQuery: ''
    }
  }

  handleInput = (e) => {
    this.setState({
      searchQuery: e.target.value
    })
  }

  findMatchingStudio = (query) => {
    return this.props.studios.filter((studio) => { 
      let studioName = studio.name.toUpperCase();
      return studioName.match(query);
   })
  }

  handleInputChange = (e) => {
    e.preventDefault()
    let userInput = this.state.searchQuery.trim().toUpperCase();
    let result = this.findMatchingStudio(userInput);

    this.props.storeRendered(result)
  }

  displayAllStudios = (e) => {
    e.preventDefault()
    this.props.storeRendered(this.props.studios)
  }

  priceFilter = () => {
    let ordered = this.props.studios.sort((a, b) => (a.dropInFee > b.dropInFee) ? 1 : -1);
    let lastElement = ordered.pop();
    return ordered.unshift(lastElement);
  }

  sortPrice = (e) => {
    e.preventDefault()
    let filteredPrice = this.priceFilter();
    
    this.props.storeRendered(filteredPrice);
  }

  render() {
    return (
      <section className="searchStudioContainer">
        <form className="searchStudio" onSubmit={this.handleInputChange}>
          <input 
            type='text'
            placeholder="Search By Studio Name"
            value={this.state.searchQuery}
            onChange={this.handleInput}
            className="studioSearch" 
          />
        </form>
          {/* Buttons below need to be moved into Controls.js */}
          <section className='btnContainer'>
            <input 
            value='Show all studios'
            type='submit'
            onClick={this.displayAllStudios}
            className='showAll'
            />
            <input
            value='Sort by price' 
            type='submit'
            onClick={this.sortPrice}
            className='sort'
            />
          </section>
      </section>
    );
  }
}

export default Search;