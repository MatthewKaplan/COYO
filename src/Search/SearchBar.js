import React from "react";
import Search from "./Search";

const SearchBar = (props) => {
  return (
    <section className="SearchBar">
      <Search 
        updateResults={props.updateResults}/>
    </section>
  );
};

export default SearchBar;