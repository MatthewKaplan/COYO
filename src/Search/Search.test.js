import React from "react";
import Search from "./Search";
import { shallow } from "enzyme";

let mockArr = [];

let mockSearchQuery = "Whole Yoga";

let mockStudio = {
  name: "Whole Yoga",
  mainTypeOffered: 5,
  address: "1735 E 17th Ave, Denver, CO 80218",
  phoneNum: "(303)333-9642",
  openEveryday: true,
  mission: "Where body and mind become whole.",
  googleRating: 4.8,
  dropInFee: "$17.00",
  teacherTraining: true,
  link: "https://wholeyoga.com",
  img: "https://i.imgur.com/w7KfQWv.jpg"
};

let mockStudios = [
  {
    name: "Whole Yoga",
    mainTypeOffered: 5,
    address: "1735 E 17th Ave, Denver, CO 80218",
    phoneNum: "(303)333-9642",
    openEveryday: true,
    mission: "Where body and mind become whole.",
    googleRating: 4.8,
    dropInFee: "$17.00",
    teacherTraining: true,
    link: "https://wholeyoga.com",
    img: "https://i.imgur.com/w7KfQWv.jpg"
  },
  {
    name: "Rino Yoga Social",
    mainTypeOffered: 8,
    address: "3101 Walnut St, Denver, CO 80205",
    phoneNum: "(720)724-8393",
    openEveryday: true,
    mission: "At RYS, our mission is simple: make yoga accessible for all.",
    googleRating: 5,
    dropInFee: "Suggested donation",
    teacherTraining: false,
    link: "https://www.rinoyogasocial.com/",
    img: "https://i.imgur.com/nKIbWP2.jpg"
  }
];

const mockHandlInputChange = jest.fn();
const mockStoreRendered = jest.fn();

describe("Search", () => {
  let wrapper, instance;
  beforeEach(() => {
    wrapper = shallow(
      <Search
        studio={mockStudio}
        studios={mockStudios}
        storeRendered={mockStoreRendered}
        searchQuery={mockSearchQuery}
        handleInputChange={mockHandlInputChange}
      />
    );
    instance = wrapper.instance();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have default state", () => {
    const initialSearchQueryState = wrapper.state("searchQuery");
    expect(initialSearchQueryState).toEqual("");
  });

  describe("handleInputChange", () => {
    it("should invoke 'findMatchingStudio' with the correct param", () => {
      let MockFn = jest.spyOn(instance, "findMatchingStudio");
      instance.handleInputChange({ preventDefault: () => {} });
      expect(MockFn).toHaveBeenCalled();
    });

    it("should invoke 'storeRendered' with the correct params", () => {
      instance.handleInputChange({
        preventDefault: () => {}
      });
      expect(mockStoreRendered).toHaveBeenCalledWith(mockStudios);
    });
  });

  it("should render an input", () => {
    const searchInput = wrapper.find(".studioSearch");
    expect(searchInput.length).toBe(1);
  });

  it("should take text from input box to filter studios when handleInputChange is invoked", () => {
    wrapper
      .find(".studioSearch")
      .simulate(
        "keypress",
        { key: "Enter" },
        { target: { value: "Whole Yoga" } }
      );
  });

  it("should setState after handleInput is invoked", () => {
    wrapper
      .find(".studioSearch")
      .simulate("change", { target: { value: "Whole Yoga" } });
    expect(mockSearchQuery).toEqual("Whole Yoga");
  });

  it("should find a matching studio", () => {
    let result = wrapper.instance().findMatchingStudio("Rino Yoga Social");
    expect(result).toEqual(mockArr);
  });
});
