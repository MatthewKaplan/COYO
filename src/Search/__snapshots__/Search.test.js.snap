// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<section
  className="searchStudioContainer"
>
  <form
    className="searchStudio"
    data-test="search-studio"
    onSubmit={[Function]}
  >
    <input
      className="studioSearch"
      data-test="search-input"
      onChange={[Function]}
      placeholder="Search By Studio Name"
      type="text"
      value=""
    />
  </form>
</section>
`;
